core/src/main/java/com/linecorp/armeria/server/DefaultServerErrorHandler.java

```diff
@@ -66,10 +66,6 @@ public HttpResponse onServiceException(ServiceRequestContext ctx, Throwable caus
         final boolean isAnnotatedService = serviceConfig.service().as(AnnotatedService.class) != null;
         if (isAnnotatedService) {
             if (cause instanceof IllegalArgumentException) {
-                if (needsToWarn()) {
-                    logger.warn("{} Failed processing a request:", ctx, cause);
-                }
-
```

Comment:
You could also remove `needsToWarn()` because it's not used anymore.

Reply:
I will remove it. Thanks.